pkg/images/enricher/enricher_impl.go

```diff
@@ -649,17 +655,38 @@ func (e *enricherImpl) checkRegistryForImage(image *storage.Image) error {
 	return nil
 }
 
-func (e *enricherImpl) getRegistriesForContext(ctx EnrichmentContext) (registries.Set, error) {
-	registrySet := e.integrations.RegistrySet()
+func (e *enricherImpl) getRegistriesForContext(ctx EnrichmentContext) ([]registryTypes.ImageRegistry, error) {
+	registries := e.integrations.RegistrySet().GetAll()
 	if ctx.Internal {
-		return registrySet, nil
+		if ctx.Source == nil {
```

Comment:
The `then`-block might be worth to factor out into its own function, as the readability is suffering from the indentation level